### PR TITLE
import: multiple wordpress attachments may point to the same URL, download it only once

### PR DESCRIPTION
**Pull request by catdog2**
This PR also includes a very small fix to support URLs with non-ascii characters which is unrelated but touches the same lines.

### CODE REVIEW

File: pelican/tools/pelican_import.py [hunk 1]
```diff
@@ -667,7 +667,12 @@ def download_attachments(output_path, urls):
             os.makedirs(full_path)
         print('downloading {}'.format(filename))
         try:
-            urlretrieve(url, os.path.join(full_path, filename))
+            if(url in successful_url_cache):
+                print('already downloaded: {} ... skipping'.format(filename))
+            else:
+                quote_url = quote(url, safe="%/:=&?~#+!$,;'@()*[]")
+                urlretrieve(quote_url, os.path.join(full_path, filename))
+                successful_url_cache.add(url)
```
Comment (leotrs) [inline pelican/tools/pelican_import.py]: Does every call to `urlretrieve` generate a successful url?
Is there a way to test if `urlretrieve` was actually successful and only then add it to `successful_url_cache`?

**Comment by justinmayer** [comment]
Hi Peter. Is there any chance you would be willing to rebase your pull request on current master so it can be merged?

**Comment by catdog2** [comment]
Thanks for the reminder, I will try to find some time to revisit this in the near future.

**Comment by justinmayer** [comment]
Hey Peter. If you find the time to revisit this, your contribution would be welcome. 😄

**Pull request closed without merging**